### ISSUE TITLE
The pdo_oci driver is supported but unmaintained

### ISSUE THREAD

**Issue by morozov** [doctrine/dbal#3294]
Currently, the `pdo_oci` driver is not being continuously tested and doesn't pass the existing test suite:
```
$ phpunit -c pdo-oci.phpunit.xml 
PHPUnit 7.1.4 by Sebastian Bergmann and contributors.

................................................................SSS..........   77 / 5144 (  1%)
..SS..............S.......SS...S.......SSS..........SS....SS..............SSS  154 / 5144 (  2%)
..............SS...SSS.........SS...S............S.S.........................  231 / 5144 (  4%)
..........................................EEE.....SSSS........SSEEEEEEEEEEEEE  308 / 5144 (  5%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEESSSSSSSSSSS  385 / 5144 (  7%)
SSSSSS...S.SSSSSSSSSSSSSSSSSSSSSSSSSSSS............SS.....S.SSSSSSS..........  462 / 5144 (  8%)
......FEEE................SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  539 / 5144 ( 10%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  616 / 5144 ( 11%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS........S  693 / 5144 ( 13%)
...S.....S......S.SS.........S.S............SSSS........SSSSSSSSSSSSSSSSSSSSS  770 / 5144 ( 14%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  847 / 5144 ( 16%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  924 / 5144 ( 17%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS 1001 / 5144 ( 19%)
SSSSSSSSSSSSSSSSSSSSSSSSSSSFFE.F.................SSS..SS.SSSSSSSSSSS.....EEEE 1078 / 5144 ( 20%)
E.....E..........SEE...E..................................................... 1155 / 5144 ( 22%)
................................................................I........SSS. 1232 / 5144 ( 23%)
......................S...................................................... 1309 / 5144 ( 25%)
............................................................................. 1386 / 5144 ( 26%)
....S........................................................................ 1463 / 5144 ( 28%)
............................................................................. 1540 / 5144 ( 29%)
.........S................................................................... 1617 / 5144 ( 31%)
............................................................................. 1694 / 5144 ( 32%)
.............S............................................................... 1771 / 5144 ( 34%)
............................................................................. 1848 / 5144 ( 35%)
...................I........SSS.............................................. 1925 / 5144 ( 37%)
............................................................................. 2002 / 5144 ( 38%)
..................................................................S.......... 2079 / 5144 ( 40%)
..SSS........................................................................ 2156 / 5144 ( 41%)
............................................................................. 2233 / 5144 ( 43%)
.......................................S.............SSS..................... 2310 / 5144 ( 44%)
............................................................................. 2387 / 5144 ( 46%)
............................................................................. 2464 / 5144 ( 47%)
.............S............SSS................................................ 2541 / 5144 ( 49%)
............................................................................. 2618 / 5144 ( 50%)
...............................................................S............S 2695 / 5144 ( 52%)
SS........................................................................... 2772 / 5144 ( 53%)
............................................................................. 2849 / 5144 ( 55%)
.................................S.............SSS......................S.... 2926 / 5144 ( 56%)
............................................................................. 3003 / 5144 ( 58%)
.........................................................................I... 3080 / 5144 ( 59%)
.....SSS....................S......S......................................... 3157 / 5144 ( 61%)
............................................................................. 3234 / 5144 ( 62%)
.................................I........SSS......................S.....S... 3311 / 5144 ( 64%)
............................................................................. 3388 / 5144 ( 65%)
.......................................................................I..... 3465 / 5144 ( 67%)
...SSS....................................................................... 3542 / 5144 ( 68%)
............................................................................. 3619 / 5144 ( 70%)
..........................I........SSS....................................... 3696 / 5144 ( 71%)
............................................................................. 3773 / 5144 ( 73%)
......................................................I........SSS........... 3850 / 5144 ( 74%)
............................................................................. 3927 / 5144 ( 76%)
............................................................................. 4004 / 5144 ( 77%)
..........I........SSS..........................................S......S..S.S 4081 / 5144 ( 79%)
............................................................................. 4158 / 5144 ( 80%)
.............................................I........SSS.................... 4235 / 5144 ( 82%)
......................II..................................................... 4312 / 5144 ( 83%)
.........SSS...................S.....S........S.............................. 4389 / 5144 ( 85%)
............................................................................. 4466 / 5144 ( 86%)
............................................................................. 4543 / 5144 ( 88%)
....................................R.......................................S 4620 / 5144 ( 89%)
............................................................................. 4697 / 5144 ( 91%)
............................................................................. 4774 / 5144 ( 92%)
...........................................................................S. 4851 / 5144 ( 94%)
.SSS......................................................................... 4928 / 5144 ( 95%)
............................................................................. 5005 / 5144 ( 97%)
............................................................................. 5082 / 5144 ( 98%)
..............................................................                5144 / 5144 (100%)
```
Depending on the nature of those failures and the actual stability of the `pdo_oci` extension, we need to either add the driver to the build matrix, or drop its support.

**Comment by morozov** [comment]
The [list](https://doc.continuousphp.com/php/7.1.0/) of available PHP extensions on ContinuousPHP doesn't contain either of `oci8` or `pdo_oci`, however, we use the former. Once it's clear if we want to support `pdo_oci`, I'll contact their support and see if `pdo_oci` also available or we can compile it ourselves.

**Comment by morozov** [comment]
Fixed by #3294.
